logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/Message.kt

```diff
@@ -304,8 +305,17 @@ sealed interface Message {
         }
     }
 
-    enum class Status {
-        PENDING, SENT, READ, FAILED, FAILED_REMOTELY
+    sealed class Status {
+        object Pending : Status()
+        object Sent : Status()
+
+        object Delivered : Status()
+
+        object Read : Status()
+
+        object Failed : Status()
+
+        object FailedRemotely : Status()
```

Comment:
Better add `toString()` here for logging purposes.

These will look ugly in logging as we do not have `data object` yet (gotta upgrade to Kotlin 1.9)